controllers/configmap_controller.go

```diff
@@ -171,32 +175,38 @@ func (r *ConfigMapReconciler) ensureInstancesAreUpToDate(instances []*instances.
 			// that has issues with configuration.
 			return errors.Wrapf(err, "error configuring host with address %s", instance.Address)
 		}
+		r.recorder.Eventf(windowsInstances, core.EventTypeNormal, "InstanceSetup",
+			"Configured instance with address %s as a worker node", instance.Address)
 	}
 	return nil
 }
 
 // deconfigureInstances removes all BYOH nodes that are not specified in the given instances slice, and
 // deconfigures the instances associated with them.
 func (r *ConfigMapReconciler) deconfigureInstances(instances []*instances.InstanceInfo, nodes *core.NodeList) error {
+	windowsInstances := &core.ConfigMap{ObjectMeta: meta.ObjectMeta{Name: wiparser.InstanceConfigMap,
+		Namespace: r.watchNamespace}}
 	for _, node := range nodes.Items {
 		// Only looking at BYOH nodes
 		if _, present := node.Annotations[BYOHAnnotation]; !present {
 			continue
 		}
 		// Check for instances associated with this node
-		if hasEntry := hasAssociatedInstance(&node, instances); hasEntry {
+		if hasEntry := getAssociatedInstance(&node, instances); hasEntry {
 			continue
 		}
 		// no instance found in the provided list, remove the node from the cluster
 		if err := r.deconfigureInstance(&node); err != nil {
 			return errors.Wrapf(err, "unable to deconfigure instance with node %s", node.GetName())
 		}
+		r.recorder.Eventf(windowsInstances, core.EventTypeNormal, "InstanceTeardown",
+			"Deconfigured node with addresses %v", node.Status.Addresses)
 	}
 	return nil
 }
 
-// hasAssociatedInstance returns true if the given node is associated with an instance in the given slice
-func hasAssociatedInstance(node *core.Node, instances []*instances.InstanceInfo) bool {
+// getAssociatedInstance returns true if the given node is associated with an instance in the given slice
+func getAssociatedInstance(node *core.Node, instances []*instances.InstanceInfo) bool {
```

Comment:
Why is this change being made? The function isn't returning an instance so its not a getter.

Reply:
Sorry my bad. I thought I reverted it.